ms/apps/readms.cc

```diff
@@ -744,11 +744,11 @@ std::vector<Int64> doHDF5 (int seqnr, const String& name)
     HDF5Group hspw(hfile, groupNames[i]);
     // Get the attributes.
     Record attr (HDF5Record::readRecord (hspw, "ATTR"));
-    CountedPtr<HDF5DataSet> hdata;
-    CountedPtr<HDF5DataSet> hfloatdata;
-    CountedPtr<HDF5DataSet> hflag;
-    CountedPtr<HDF5DataSet> hweightspectrum;
-    hflag = new HDF5DataSet (hspw, "FLAG", (Bool*)0);
+    std::shared_ptr<HDF5DataSet> hdata;
+    std::shared_ptr<HDF5DataSet> hfloatdata;
+    std::shared_ptr<HDF5DataSet> hflag;
+    std::shared_ptr<HDF5DataSet> hweightspectrum;
+    hflag.reset (new HDF5DataSet (hspw, "FLAG", (Bool*)0));
```

Comment:
```suggestion
    std::shared_ptr<HDF5DataSet> hflag = std::make_shared<HDF5DataSet>(hspw, "FLAG", nullptr);
    std::shared_ptr<HDF5DataSet> hweightspectrum;
```

Reply:
done

Reply:
I see you indeed can't use `nullptr`, since there are overloads for different pointer types.